File: frontend/src/types/category.ts
export interface Category {
  id: string
  user_id: string
  name: string
  type: 'feed' | 'paper'
  position: number
  created_at: string
}

export interface CreateCategoryRequest {
  name: string
  type: 'feed' | 'paper'
}

export interface RenameCategoryRequest {
  name: string
}

export interface MoveToCategoryRequest {
  category_id: string
}

export interface ListCategoriesResponse {
  categories: Category[]
}

File: backend/internal/model/category.go
package model

// Category type constants
const (
	CategoryTypeFeed  = "feed"
	CategoryTypePaper = "paper"
)

// Category represents a user-defined category for feeds or papers
type Category struct {
	Base
	UserID   string `gorm:"type:varchar(36);not null;uniqueIndex:idx_user_cat_name" json:"user_id"`
	Name     string `gorm:"type:varchar(100);not null;uniqueIndex:idx_user_cat_name" json:"name"`
	Type     string `gorm:"type:varchar(20);not null;uniqueIndex:idx_user_cat_name" json:"type"`
	Position int    `gorm:"default:0" json:"position"`
	User     *User  `gorm:"foreignKey:UserID" json:"user,omitempty"`
}
